server/src/test/java/org/elasticsearch/update/UpdateIT.java

```diff
@@ -225,7 +225,7 @@ public void testUpsertDoc() throws Exception {
         UpdateResponse updateResponse = client().prepareUpdate(indexOrAlias(), "type1", "1")
                 .setDoc(XContentFactory.jsonBuilder().startObject().field("bar", "baz").endObject())
                 .setDocAsUpsert(true)
-                .setFields("_source")
+                .setFetchSource(true)
```

Comment:
Hmmm. I *think* this is true by default. Can you check what happens if you drop the `.setFetchSource(true)` part?

Reply:
By default the field ```fetchSource``` is null in the class UpdateRequest:
```
    private FetchSourceContext fetchSourceContext;
```. 
If we drop this, we'll get a AssertionError.

Reply:
I see it. I was tired when I was reading it. You are right to do what you did.